Applications/xia2_main.py

```diff
@@ -63,6 +65,22 @@ def check_environment():
         )
 
 
+def check_hdf5_master_files(master_files):
+    """Check the input HDF5 master files look a little bit like HDF5 master
+    files and not just the data files: if the latter then sys.exit() with a
+    helpful message"""
+
+    bad = []
+
+    for filename in master_files:
+        with h5py.File(filename, "r") as f:
+            if b"/data" in f and b"/entry" not in f:
+                bad.append(filename)
+
+    if bad:
+        sys.exit("Not master files: %s" % " ".join(bad))
```

Comment:
I think this error message probably needs to be clearer still. Something along the lines of "You appear to have passed a HDF5 data file rather than master file as input." Maybe also including a hint for what we are expecting instead, i.e. `*_master.h5` or `*.nxs`
Would it be possible to at least look for related master files and point the user towards them (even if we don't automatically use the master file)?

Reply:
@rjgildea like this? 

```
Command line: xia2 image=/Users/graeme/data/i04-eiger-small/Therm_6_2_000001.h5
Provided input files not master files:
  /Users/graeme/data/i04-eiger-small/Therm_6_2_000001.h5
do you mean one of:
  /Users/graeme/data/i04-eiger-small/Therm_6_1_master.h5
  /Users/graeme/data/i04-eiger-small/Therm_6_2_master.h5
or:
  /Users/graeme/data/i04-eiger-small/Therm_6_2.nxs
  /Users/graeme/data/i04-eiger-small/Therm_6_1.nxs
```

Done in last commit 🙂